source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1514,7 +1514,11 @@ servers. For a "node is recovering" error, single-threaded clients MUST NOT
 mark the topology as "stale". If a node is recovering for some time, an
```

Comment:
In [updateRSFromPrimary](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#updatersfromprimary)  it says:

> The client SHOULD clear its connection pool for the old primary, too: the connections are all bad because the old primary has closed its sockets.

Does this also need to be updated to say that on >=4.2 we should not clear the pool?

Reply:
I was not referring to this section where the comment is located. I was referring to the [updateRSFromPrimary](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#updatersfromprimary) section. (Github PRs don't let you put a comment on lines that haven't changed.)

So carrying on with my question about updateRSFromPrimary. According to the updated SDAM spec in this PR, drivers will *still* clear the pool when a primary steps down; when the new primary reports itself via a heartbeat, SDAM says that the connections to the old primary SHOULD be closed.

Personally I think that we should not change the SDAM spec in this PR and we should say that drivers still close connections when a primary steps down. Cursors will still survive elections which is the main motivation of this project.

Reply:
One of the cases I'm considering in this thread is when a election occurs when an application is idle. The sequence of events would be:
1. Primary is P1
1. Primary steps down and new primary, P2, is elected.
1. SDAM heartbeat for P2 returns and reports that P2 is the primary.
1. SDAM invalidates the old primary P1 and clears its connection pool.
1. Connections have not survived primary step down.

Reply:
Any thoughts on this?